junit-platform-surefire-provider/src/test/java/org/junit/platform/surefire/provider/TestPlanScannerFilterTests.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.platform.surefire.provider;
+
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.platform.launcher.core.LauncherFactory;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+class TestPlanScannerFilterTests {
+
+	@Test
+	void emptyClassNotAccepted() {
+		assertFalse(newFilter().accept(EmptyClass.class));
+	}
+
+	@Test
+	void classWithMethodsIsNotAccepted() {
+		assertFalse(newFilter().accept(ClassWithMethods.class));
+	}
+
+	@Test
+	void classWithTestMethodsIsAccepted() {
+		assertTrue(newFilter().accept(ClassWithTestMethods.class));
+	}
+
+	@Test
+	void classWithNestedTestClassIsAccepted() {
+		assertTrue(newFilter().accept(ClassWithNestedTestClass.class));
+	}
+
+	@Test
+	void classWithDeeplyNestedTestClassIsAccepted() {
+		assertTrue(newFilter().accept(ClassWithDeeplyNestedTestClass.class));
+	}
+
+	@Test
+	@Disabled("Methods annotated with @TestFactory are not accepted, they are containers, not tests")
```

Comment:
Classes which contain only `@TestFactory` annotated methods do not seem to be discovered by surefire provider. This condition https://github.com/junit-team/junit5/blob/master/junit-platform-surefire-provider/src/main/java/org/junit/platform/surefire/provider/TestPlanScannerFilter.java#L38 accepts only test methods and `@TestFactory` methods are containers.

Please let me know if it is valuable to keep these disabled tests here or not.